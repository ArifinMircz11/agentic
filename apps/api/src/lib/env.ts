import 'dotenv/config'

import { z } from 'zod'

export const envSchema = z.object({
  NODE_ENV: z
    .enum(['development', 'test', 'production'])
    .default('development'),
  DATABASE_URL: z.string().url(),
  JWT_SECRET: z.string()
})

// eslint-disable-next-line no-process-env
export const env = envSchema.parse(process.env)

export const isProd = env.NODE_ENV === 'production'
