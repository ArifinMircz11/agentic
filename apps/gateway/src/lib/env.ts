import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import {
  envSchema as baseEnvSchema,
  parseEnv as parseBaseEnv
} from '@agentic/platform-hono'
import { z } from 'zod'

export const envSchema = baseEnvSchema
  .extend({
    AGENTIC_API_BASE_URL: z.string().url(),
    AGENTIC_API_KEY: z.string().nonempty(),

    DO_RATE_LIMITER: z.custom<DurableObjectNamespace>(
      (ns) => ns && typeof ns === 'object'
    )
  })
  .strip()
export type RawEnv = z.infer<typeof envSchema>
export type Env = Simplify<ReturnType<typeof parseEnv>>

export function parseEnv(inputEnv: Record<string, unknown>) {
  const baseEnv = parseBaseEnv({
    SERVICE: 'gateway',
    ...inputEnv
  })

  const env = parseZodSchema(
    envSchema,
    { ...inputEnv, ...baseEnv },
    {
      error: 'Invalid environment variables'
    }
  )

  return {
    ...baseEnv,
    ...env
  }
}
