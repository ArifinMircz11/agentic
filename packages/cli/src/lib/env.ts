export const agenticApiBaseUrl =
  process.env.AGENTIC_API_BASE_URL || 'https://api.agentic.so'
