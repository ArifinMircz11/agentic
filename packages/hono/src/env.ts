import 'dotenv/config'

import type { Simplify } from 'type-fest'
import { parseZodSchema } from '@agentic/platform-core'
import { z } from 'zod'

import { logLevelsSchema } from './logger'

export const envSchema = z
  .object({
    ENVIRONMENT: z
      .enum(['development', 'test', 'production'])
      .default('development'),

    SERVICE: z.enum(['api', 'gateway']),

    LOG_LEVEL: logLevelsSchema.default('info'),

    SENTRY_DSN: z.string().url().optional()
  })
  .strip()

export function parseEnv(inputEnv: unknown) {
  const env = parseZodSchema(envSchema, inputEnv, {
    error: 'Invalid environment variables'
  })

  const isDev = env.ENVIRONMENT === 'development'
  const isTest = env.ENVIRONMENT === 'test'
  const isProd = env.ENVIRONMENT === 'production'
  const isBrowser = (globalThis as any).window !== undefined

  if (isProd && !env.SENTRY_DSN) {
    throw new Error(
      'Internal error: missing required "SENTRY_DSN" environment variable in production'
    )
  }

  return {
    ...env,
    isDev,
    isTest,
    isProd,
    isBrowser
  }
}

export type Env = Simplify<ReturnType<typeof parseEnv>>
