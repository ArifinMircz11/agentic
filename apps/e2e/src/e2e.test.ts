import contentType from 'fast-content-type-parse'
import defaultKy from 'ky'
import { describe, expect, test } from 'vitest'

import { env } from './env'
import { fixtureSuites } from './fixtures'

const ky = defaultKy.extend({
  prefixUrl: env.AGENTIC_GATEWAY_BASE_URL,

  // Disable automatic retries for testing.
  retry: 0
})

for (const [i, fixtureSuite] of fixtureSuites.entries()) {
  const { title, fixtures } = fixtureSuite

  describe(title, () => {
    let responseBody: any | undefined

    for (const [j, fixture] of fixtures.entries()) {
      const method = fixture.request?.method ?? 'GET'
      const {
        status = 200,
        contentType: expectedContentType = 'application/json',
        headers: expectedHeaders,
        body: expectedBody
      } = fixture.response ?? {}
      const { snapshot = status >= 200 && status < 300 } =
        fixture.response ?? {}

      test.sequential(
        `${i}.${j}: ${method} ${fixture.path}`,
        {
          timeout: fixture.timeout ?? 60_000
        },
        // eslint-disable-next-line no-loop-func
        async () => {
          const res = await ky(fixture.path, fixture.request)
          expect(res.status).toBe(status)

          const { type } = contentType.safeParse(
            res.headers.get('content-type') ?? ''
          )
          expect(type).toBe(expectedContentType)

          if (expectedHeaders) {
            for (const [key, value] of Object.entries(expectedHeaders)) {
              expect(res.headers.get(key)).toBe(value)
            }
          }

          let body: any

          if (type.includes('json')) {
            try {
              body = await res.json()
            } catch (err) {
              console.error('json error', err)
              throw err
            }
          } else if (type.includes('text')) {
            body = await res.text()
          } else {
            body = await res.arrayBuffer()
          }

          if (expectedBody) {
            expect(body).toEqual(expectedBody)
          }

          if (snapshot) {
            expect(body).toMatchSnapshot()
          }

          if (status >= 200 && status < 300) {
            if (!responseBody) {
              responseBody = body
            } else {
              expect(body).toEqual(responseBody)
            }
          }

          console.log(`${i}.${j}: ${method} ${fixture.path}`, {
            status,
            body,
            headers: Object.fromEntries(res.headers.entries())
          })
        }
      )
    }
  })
}
