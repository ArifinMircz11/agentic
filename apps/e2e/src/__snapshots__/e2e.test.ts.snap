// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic OpenAPI getPost(1) > 0.0: GET dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`Basic OpenAPI getPost(1) > 0.1: GET dev/test-basic-openapi@b6e21206/getPost?postId=1 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`Basic OpenAPI getPost(1) > 0.2: GET dev/test-basic-openapi@b6e21206/get_post?postId=1 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`Basic OpenAPI getPost(1) > 0.3: GET dev/test-basic-openapi@b6e21206/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`Basic OpenAPI getPost(1) > 0.4: POST dev/test-basic-openapi/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;

exports[`Basic OpenAPI getPost(1) > 0.5: POST dev/test-basic-openapi@latest/getPost 1`] = `
{
  "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "userId": 1,
}
`;
